var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -470,6 +470,12 @@ def f77(self):
             result = os.path.join(self.spec.prefix.bin, 'flang')
         return result
 
+    @property
+    def libs(self):
+        shared = '+shared_libs' in self.spec
+        return find_libraries('libLLVMCore', self.prefix, shared=shared,
```

Comment:
This seems a bit arbitrary to me; there are tons of libraries:

```
$ ./llvm-config --libnames
libLLVMWindowsManifest.a libLLVMXRay.a libLLVMLibDriver.a libLLVMDlltoolDriver.a libLLVMCoverage.a libLLVMLineEditor.a libLLVMX86Disassembler.a libLLVMX86AsmParser.a libLLVMX86CodeGen.a libLLVMX86Desc.a libLLVMX86Info.a libLLVMOrcJIT.a libLLVMMCJIT.a libLLVMJITLink.a libLLVMOrcTargetProcess.a libLLVMOrcShared.a libLLVMInterpreter.a libLLVMExecutionEngine.a libLLVMRuntimeDyld.a libLLVMSymbolize.a libLLVMDebugInfoPDB.a libLLVMDebugInfoGSYM.a libLLVMOption.a libLLVMObjectYAML.a libLLVMMCA.a libLLVMMCDisassembler.a libLLVMLTO.a libLLVMPasses.a libLLVMCFGuard.a libLLVMCoroutines.a libLLVMObjCARCOpts.a libLLVMHelloNew.a libLLVMipo.a libLLVMVectorize.a libLLVMLinker.a libLLVMInstrumentation.a libLLVMFrontendOpenMP.a libLLVMFrontendOpenACC.a libLLVMExtensions.a libLLVMDWARFLinker.a libLLVMGlobalISel.a libLLVMMIRParser.a libLLVMAsmPrinter.a libLLVMDebugInfoDWARF.a libLLVMSelectionDAG.a libLLVMCodeGen.a libLLVMIRReader.a libLLVMAsmParser.a libLLVMInterfaceStub.a libLLVMFileCheck.a libLLVMFuzzMutate.a libLLVMTarget.a libLLVMScalarOpts.a libLLVMInstCombine.a libLLVMAggressiveInstCombine.a libLLVMTransformUtils.a libLLVMBitWriter.a libLLVMAnalysis.a libLLVMProfileData.a libLLVMObject.a libLLVMTextAPI.a libLLVMMCParser.a libLLVMMC.a libLLVMDebugInfoCodeView.a libLLVMDebugInfoMSF.a libLLVMBitReader.a libLLVMCore.a libLLVMRemarks.a libLLVMBitstreamReader.a libLLVMBinaryFormat.a libLLVMTableGen.a libLLVMSupport.a libLLVMDemangle.a
```

And when `+llvm_dylib` it should `libLLVM-[version].so`, which may or may not include a version suffix if `version_suffix` is set.

Reply:
If `~llvm_dylib` is used, do we end up with a single interface library `libLLVM.a`?

Reply:
> This seems a bit arbitrary to me; there are tons of libraries:

It was not done arbitrarily. I built a stripped down version of llvm and looked at the libraries. I also compared libraries to other llvm builds with variants that generated more libraries to make sure of the ones always present. There is an assumption that "core" means what I think most people interpret it to mean.

Reply:
> If `~llvm_dylib` is used, do we end up with a single interface library `libLLVM.a`?

No, if `+llvm_dylib`, there is a `libLLVM.so` though.

Reply:
Can you special case the +llvm_dylib case to only look for `libLLVM.{so,dylib}`? That would be the expected library to return. And on ~llvm_dylib, I think we can just return the full list of libLLVM*.{so,dylib,a}.

Reply:
Let's just ask `llvm-config`, then we don't have to deal with variants at all.

Reply:
Don't expect llvm-config to work well 😆 iirc it was not returnung sensible libs with +llvm_dylib. Also it would assume it's compiled for the host